qutebrowser/browser/commands.py

```diff
@@ -775,19 +779,23 @@ def _to_close(i):
                         (prev and i < cur_idx) or
                         (next_ and i > cur_idx))
 
+        if force:
+            message.warning("--force is deprecated, use --pinned close instead.")
+            pinned = 'close'
+
         # close as many tabs as we can
         first_tab = True
         pinned_tabs_cleanup = False
         for i, tab in enumerate(self._tabbed_browser.widgets()):
             if _to_close(i):
-                if force or not tab.data.pinned:
+                if pinned == 'close' or not tab.data.pinned:
                     self._tabbed_browser.close_tab(tab, new_undo=first_tab)
                     first_tab = False
                 else:
                     pinned_tabs_cleanup = tab
 
         # Check to see if we would like to close any pinned tabs
-        if pinned_tabs_cleanup:
+        if pinned_tabs_cleanup and pinned == 'prompt':
             self._tabbed_browser.tab_close_prompt_if_pinned(
```

Comment:
In the lambda below this line (GitHub won't let me comment there as it's not part of your changes), `force=True` is still used, thus triggering the warning (and failing `test_pinned_tabonly_prompt_yes`). If you change that to `pinned='close'` instead, I think everything should work fine.